### DISCUSSION TITLE
Access Forbidden?

### DISCUSSION THREAD

**Discussion by virgilejarrige** [UtrechtUniversity/yoda#386] [Q&A] [answered]
Hello,

Thank you for sharing the source code of yoda and for this "easy to deploy" test environment! It is really well documented! Awesome work!

I used the docker-compose.yml that you provided on github to test yoda, and here it is: https://portal-yoda-test.scienceouverte.unistra.fr:8443/

When I'm logged in as a researcher, it won't let me start a deposit. The error message "network issue" blinks and then "Access forbidden" appears.

Any idea where it could come from?

Thank you in advance for your help!

Take care,

Virgile

​​Virgile Jarrige 
virgile.jarrige@unistra.fr

Pôle Calcul Et Services Avancés à la Recherche (CESAR)
Département Services Métiers 
Direction du Numérique
Université de Strasbourg
https://www.unistra.fr/

**Comment by stsnel** [comment, marked as answer]
Hello Virgile,

Thanks for the issue report and great to hear you are trying out Yoda!

The root cause of the issue was that the Docker setup currently does not have support for OpenSearch, which is used by the Deposit module. We've fixed a configuration issue that resulted in the Deposit module not working properly in the Docker setup because of this. If you pull the newest version of the Docker setup from the development branch, you should be able to start a deposit.

The search functionality of the Deposit module is still not available in the Docker setup, considering that it needs OpenSearch. I will discuss in our team if/when we can add this to the Docker setup.

Best regards,
Sietse

Yoda development team

**Reply by virgilejarrige** [reply]
Hello Sietse,

Thank you for your quick answer!

I did read that the search functionality does not work yet with the docker setup, but that was not the feature that we wanted to test yet. :)

I pulled the new version but the same error message appears when I try to do a new deposit. Maybe that I am doing something wrong?

I am using the development branch and it seems up to date
`debian@yoda:~/yoda$ git branch`
`* development`
`debian@yoda:~/yoda$ git pull`
`Already up to date.`

I modified a little bit "up.sh" so it starts the containers as a daemon and uses "compose" from docker and not "docker-compose" 
`debian@yoda:~/yoda$ cat docker/up.sh `
`#!/bin/bash`
`docker compose up -d "$@"`

I then use this little script to start everything
`debian@yoda:~/yoda$ cd docker/compose`
`debian@yoda:~/yoda/docker/compose$ ../up.sh`

And all is starting normally.

When I look into the "docker logs", 
- eus and irods have some error messages but seem ok
- db and extuserdb talk about duplicates that can not be created, but i guess it's normal because  the database is already there
- the others are working without any error message

I tried to look into the logs of the containers, but didn't find anything.

Any idea?

Best regards,

Virgile

**Reply by stsnel** [reply]
Hi Virgile,

The procedure for updating the Docker setup involves a few additional steps:

You'll also need to pull the latest images:

```
docker compose pull
```

And then re-start the setup after removing the old volume data.

```
docker compose down -v
docker compose up
```

Removing old volume data ensures that (among other things) configuration files get updated automatically.

If you prefer to preserve volume data, you could also perform this change manually without removing data. In that case, you would need to set `enable_open_search` in `/etc/irods/yoda-ruleset/rules_uu.cfg` in the `provider.yoda` container to `'false'` and `OPEN_SEARCH_ENABLED` in `/var/www/yoda/flask.cfg` in the `portal.yoda` container to `False`. After updating the portal configuration manually, you'll also need to trigger a config reload in the portal container using `touch /var/www/yoda/*.wsgi`.

Best regards,
Sietse

**Reply by virgilejarrige** [reply]
"docker compose pull" was, indeed, the missing part on my side! 

It's now working, thanks a bunch! 

Take care,

Virgile

**Reply by stsnel** [reply]
That is great to hear, thanks for your feedback :+1: 

Best regards,
Sietse
